### ISSUE TITLE
MySQL validation can fail on text that can be found in other areas

### ISSUE THREAD

**Issue by djavorszky** [djavorszky/ddn#35]
Currently, MySQL dump validation searches for the following strings in the dump:
"create database", "CREATE DATABASE", "use ", "USE ", "drop database", "DROP DATABASE". If it finds any of them, it will:
1. Remove them if only one is found
2. Report error if more than one is found.

"use " is a special case, because it is entirely possible to be contained in text. In one particular dump, there are 546 instances of the string, however none of them give instructions to MySQL itself. There's even an instance of use \` (with a backtick).

Question: What should we do with it? Validating the dump takes considerable time, as the dump has to be traversed 6 times at least. For a 5.3 G mysql dump, it took 70 seconds for 3 iterations to complete (create, CREATE, and finally "use "), at which point it failed.

I think we have the following options:
1. Create regex magic - though this would slow the search down even more
2. Remove the "use " and "USE " statements from the validation.
3. Stop validating altogether - As the importing user only has roles on the database created for it, any mysql commands that operates on other database would fail anyhow. Not to mention, we don't validate Postgresql or oracle (afaik)

What do you think @sorin-pop, @zsoltbalogh

**Comment by zsoltbalogh** [comment]
70 seconds or more is perfectly fine. We should do everything possible to avoid manual work as it's way more costly. I think it would take more time to open the file and remove the lines manually. :)

**Comment by djavorszky** [comment]
Well.. I guess I could use the following regex:
`(use |USE )['"`][a-zA-Z0-9-_]+['"`];`

Which matches:
```
use 'database';
use `database`;
use "database";

USE 'database';
USE `database`;
USE "database";

use 'database_123';
use `database_123`;
use "database_123";

USE 'database-123';
USE `database-123`;
USE "database-123";
```

Question is, how fast / slow this'll be. I'll do some tests with it.

**Comment by zsoltbalogh** [comment]
It's enough if you filter for USE<space> and use<space>. IMHO lines in the dump always start with a command.

**Comment by djavorszky** [comment]
The problem with that is that mysql allows spaces before the command. So "    use `db`;" is a valid command.

I could match any number of spaces and tabs before the use though, like so:
`[^\t|^ ]*(use |USE )+`

**Comment by djavorszky** [comment]
As discussed with @zsoltbalogh, we'll only check if lines starts with the mentioned commands:
```
create database
CREATE DATABASE
"use " (without quotes)
"USE " (without quotes)
drop database
DROP DATABASE
```

In all other scenarios the dumps would fail and should be handled on a case-by-case basis.

**Comment by djavorszky** [comment]
Updated / implemented in 0c1b7d71de117c35538019847eeee2b912dfe8aa
